include/oneapi/tbb/detail/_task_handle.h

```diff
@@ -116,6 +193,104 @@ inline bool operator!=(std::nullptr_t, task_handle const& th) noexcept {
     return th.m_handle != nullptr;
 }
 
+#if __TBB_PREVIEW_TASK_GROUP_EXTENSIONS
+class task_completion_handle {
+public:
+    task_completion_handle() : m_task_state(nullptr) {}
+
+    task_completion_handle(const task_completion_handle& other) 
+        : m_task_state(other.m_task_state)
+    {
+        // Register one more co-owner of the dynamic state
+        if (m_task_state) m_task_state->reserve();
+    }
+    task_completion_handle(task_completion_handle&& other)
+        : m_task_state(other.m_task_state)
+    {
+        other.m_task_state = nullptr;
+    }
+
+    task_completion_handle(const task_handle& th)
+        : m_task_state(th.m_handle->get_dynamic_state())
+    {
+        __TBB_ASSERT(th, "Construction of task_completion_handle from an empty task_handle");
+        // Register new co-owner of the dynamic state
+        m_task_state->reserve();
+    }
+
+    ~task_completion_handle() {
+        if (m_task_state) m_task_state->release();
+    }
+
+    task_completion_handle& operator=(const task_completion_handle& other) {
+        if (this != &other) {
```

Comment:
Perhaps `*this != other` can be used instead, or maybe direct pointer comparison of `m_task_state` with that of `other`.